healme.lic

```diff
@@ -2,198 +2,320 @@
-    Flags.reset('hm-partial-heal')
-    cast?("cast #{part} #{internal ? 'internal' : ''}")
-    pause 0.5
-    echo "partial: #{Flags['hm-partial-heal']}" if UserVars.healme_debug
-    Flags['hm-partial-heal']
-  end
-
-  def heal_wound(spells, part, internal)
-    if @using_waggle
-      spells['Heal Wounds']['cast'] = "cast #{part}"
-      spells['Heal Wounds']['cast'] = "cast #{part} internal" if internal
-      check_discern(spells['Heal Wounds'], @settings) if spells['Heal Wounds']['use_auto_mana']
-      loop do
-        Flags.reset('hm-partial-heal')
-        DRCA.cast_spell(spells['Heal Wounds'], @settings, true)
-        echo "partial: #{Flags['hm-partial-heal']}" if UserVars.healme_debug
-        break unless Flags['hm-partial-heal']
-      end
-    else
-      pause 0.5 while cast_spell('HW', @settings.empath_healing['HW'], part, internal)
+  # Get wounds inferred from HEALTH command.
+  def get_self_health
+    echo 'Checking for visible wounds, bleeders, parasites, and lodged items' if $debug_mode_hm
+    self_health = DRCH.check_health
+    # Exclude bleeders that are '(tended)' because
+    # they aren't causing immediate loss of vitality
+    # and, more technically, once the body part is
+    # healed the wound will still be listed in HEALTH
+    # as '(tended)' giving the impression you have a bleeder
+    # when in reality you don't. We have to wait for the
+    # bandages to fall off or removed explicitly in order
+    # to no longer see a bleeder in the HEALTH output.
+    # Without this workaround, the script loops repeatedly
+    # trying to heal a body part that's not injured.
+    self_health['bleeders'].each do |severity, wounds|
+      wounds.select! { |wound| wound.bleeding_rate != '(tended)' }
     end
+    # Now, remove any keys from the map if no wounds left after filtering.
+    self_health['bleeders'].select! { |severity, wounds| wounds.length > 0 }
+    remove_wounds_to_keep(self_health)
   end
 
-  def heal_scar(spells, part)
-    if @using_waggle
-      spells['Heal Scars']['cast'] = "cast #{part}"
-      check_discern(spells['Heal Scars'], @settings) if spells['Heal Scars']['use_auto_mana']
-      loop do
-        Flags.reset('hm-partial-heal')
-        DRCA.cast_spell(spells['Heal Scars'], @settings, true)
-        echo "partial: #{Flags['hm-partial-heal']}" if UserVars.healme_debug
-        break unless Flags['hm-partial-heal']
+  # Get wounds inferred from PERCEIVE HEALTH SELF.
+  def get_perc_health
+    return if @settings.permashocked
+    echo 'Perceiving internal and external wounds and scars' if $debug_mode_hm
+    remove_wounds_to_keep(DRCH.perceive_health)
+  end
+
+  # Takes a pulse on vitality then takes action accordingly.
+  # Designed to be called frequently so that low vitality
+  # and bleeding can be addressed immediately.
+  def attend_vitals
+    heal_vitality if health < [50, @settings.health_threshold].max
+    stop_bleeding if bleeding?
+    regenerate if @spells['Regenerate']
```

Comment:
Probably shouldn't regen if they have wounds they want to keep.

Reply:
Good catch